frontend/components/JoinGraveBtn.tsx

```diff
@@ -181,51 +284,39 @@ export default function JoinGraveBtn({
       return err;
     }
     if (!graveVault) {
-      // 2. Create a vault for the UP. (if needed)
+      // 2.A. Create a vault for the UP in batch transaction. (if needed)
       try {
-        const vaultTranx = await createUpVault(signer);
-        vaultAddress = vaultTranx.contractAddress;
+        const batchJoinTrax = await batchJoin(provider, signer);
+        if (!batchJoinTrax.vaultAddress) {
+          // todo test flow
+          throw new Error('There was a problem setting up your Vault.');
+        }
         // add the vault to the provider store
-        addGraveVault(vaultAddress);
+        addGraveVault(batchJoinTrax.vaultAddress);
         setJoiningStep(2);
         console.log('step 2');
       } catch (err: any) {
         handleError(err);
         return err;
       }
-      // 3. Set the vault in the forwarder contract
+    } else {
+      console.log('batch join skipped, vault already exists');
+      //2.B. Enable grave to keep assets inventory (this done in 2.A too but as part of a batch call)
```

Comment:
we should never end up here @Ledespana. in fact if we do end up here now that would mean something is broken because grave was created but its URD was not set which should be impossible